core/src/modules/locks.ts

```diff
@@ -41,3 +41,35 @@ export async function waitForLock(
 
   return { releaseLock, attempts };
 }
+
+export type getLockResponse = {
+  releaseLock: Function;
+  isLocked: boolean;
+  lockedBy: string;
+};
+
+export async function getLock(
+  key: string,
+  requestId: string = uuid.v4(),
+  ttl = LOCK_DURATION_MS,
+  lockType?: string
```

Comment:
for pattern matching, it might be best to give this a default, like `lockType = 'lock'`, this way all the lock keys will have the same number of "parts", eg: `grouparoo:lock:{type}:key` if we need to find or clear them in the future